openmeter/ent/schema/subscription_patch.go

```diff
@@ -0,0 +1,212 @@
+package schema
+
+import (
+	"entgo.io/ent"
+	"entgo.io/ent/dialect"
+	"entgo.io/ent/dialect/entsql"
+	"entgo.io/ent/schema"
+	"entgo.io/ent/schema/edge"
+	"entgo.io/ent/schema/field"
+	"entgo.io/ent/schema/index"
+
+	"github.com/openmeterio/openmeter/pkg/framework/entutils"
+)
+
+type SubscriptionPatch struct {
+	ent.Schema
+}
+
+func (SubscriptionPatch) Annotations() []schema.Annotation {
+	return []schema.Annotation{
+		entsql.Annotation{Table: "subscription_patches"},
+	}
+}
+
+func (SubscriptionPatch) Mixin() []ent.Mixin {
+	return []ent.Mixin{
+		entutils.IDMixin{},
+		entutils.NamespaceMixin{},
+		entutils.TimeMixin{},
+		entutils.MetadataAnnotationsMixin{},
+	}
+}
+
+func (SubscriptionPatch) Fields() []ent.Field {
+	return []ent.Field{
+		field.String("subscription_id").NotEmpty().Immutable(),
+		field.Time("applied_at").Immutable(),
```

Comment:
is applied_at necessary if we have created_at? Can these differ?

Reply:
Yes. `appliedAt` is programmatic time, it's kept constant through the entire use case prior creating the record and when reading the record. In practice, appliedAt will be the exact same for those in the same batch, and probably some ms / ns before `createdAt`